src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameService.cs

```diff
@@ -134,7 +135,9 @@ internal sealed class InlineRenameService : IInlineRenameService
                             }
                         }
 
-                        if (!navigationService.CanNavigateToSpan(workspace, document.Id, documentSpan.SourceSpan, cancellationToken))
+                        var canNavigate = threadingContext.JoinableTaskFactory.Run(() =>
```

Comment:
Should we/can we push the async higher? I see this method is also calling GetTextSynchronously.

Reply:
sure.  let me do that.